src/pattern_matching/myers.rs

```diff
@@ -13,75 +13,246 @@
 //! ```
 //! # extern crate itertools;
 //! # extern crate bio;
-//! use bio::pattern_matching::myers::Myers;
+//! use bio::pattern_matching::myers::Myers64;
 //! use itertools::Itertools;
 //!
 //! # fn main() {
 //! let text = b"ACCGTGGATGAGCGCCATAG";
 //! let pattern = b"TGAGCGT";
 //!
-//! let myers = Myers::new(pattern);
+//! let myers = Myers64::new(pattern);
 //! let occ = myers.find_all_end(text, 1).collect_vec();
 //!
 //! assert_eq!(occ, [(13, 1), (14, 1)]);
 //! # }
 //! ```
 
 use std::iter;
+use std::ops::Range;
 use std::u64;
+use std::borrow::Borrow;
+use std::mem::size_of;
+use std::ops::*;
+use std::marker::PhantomData;
+
+use num_traits::*;
 
 use utils::{IntoTextIterator, TextIterator, TextSlice};
+use alignment::{Alignment, AlignmentOperation, AlignmentMode};
+
+
+
+/// Integer types serving as bit vectors must implement this trait.
+pub trait Num: Add<Output=Self> + Sub<Output=Self> +
+    BitOr<Output=Self> + BitOrAssign + BitAnd<Output=Self> + BitXor<Output=Self> + Not<Output=Self> +
+    Shl<usize, Output=Self> + ShlAssign<usize> + ShrAssign<usize> +
+    Copy + Clone + PartialOrd
+{
+    fn max_value() -> Self;
+    fn zero() -> Self;
+    fn one() -> Self;
+    fn count_ones(self) -> u32;
+    fn wrapping_add(&self, other: &Self) -> Self;
+}
+
+// Implementation using num_traits types
```

Comment:
The whole idea of the `Num` trait (or however it may be called) was to enable custom implementations that don't depend on the num_traits crate, with big integers in mind. The problem is that only a single method is needed from `num_traits::int::PrimInt`, but if depending on PrimInt directly, a whole bunch of unneeded methods would need to be implemented.
However, there are other hurdles for the use of big-integer types (wrapping_add seems missing in all crates), and I have no idea about how performance would be. Therefore I'm totally ok with removing this trait and directly depending on the num_traits types.

Reply:
Yes, lets do that for now. I think it would be preferable to fix that in the num_traits crate sooner or later.